core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2617,6 +2617,22 @@ private[spark] object Utils extends Logging {
     redact(redactionPattern, kvs.toArray)
   }
 
+  def redactCommandLineArgs(conf: SparkConf, commands: Seq[String]): Seq[String] = {
+    val redactionPattern = conf.get(SECRET_REDACTION_PATTERN)
+    commands.map { cmd =>
+      var newCmd = cmd
```

Comment:
Seems we shall only match the pattern "-D$key=$value" here, otherwise return the original cmd. That way the code may be simpler.